src/main/java/com/amazon/opendistroforelasticsearch/jdbc/ResultSetImpl.java

```diff
@@ -112,15 +123,56 @@ public boolean next() throws SQLException {
         log.debug(() -> logEntry("next()"));
         checkOpen();
         boolean next = cursor.next();
+
+        if (!next && this.cursorId != null) {
+            log.debug(() -> logEntry("buildNextPageFromCursorId()"));
+            buildNextPageFromCursorId();
+            log.debug(() -> logExit("buildNextPageFromCursorId()"));
+            next = cursor.next();
```

Comment:
why call cuser.next() two times? L125?

Reply:
Because after fetching new page using cursor, a new `Cursor` object will created with `currentRow = -1`. Without doing `cursor.next()`, it will throw `IndexOutOfBoundException` for accessing Row list at index `-1`.